utils/file.c

```diff
@@ -77,27 +77,32 @@ int myst_load_file(const char* path, void** data_out, size_t* size_out)
     return ret;
 }
 
-ssize_t myst_writen(int fd, const void* data, size_t size)
+int myst_write_file_fd(int fd, const void* data, size_t size)
 {
-    ssize_t ret = 0;
+    int ret = 0;
     const uint8_t* p = (const uint8_t*)data;
     size_t r = size;
+    ssize_t n;
+
+    if (fd < 0 || !data)
```

Comment:
I think it's ok for data to be NULL, as long as `size` is 0.

Reply:
Changed.